test/Sentry.Tests/Internals/AccessModifierTests.cs

```diff
@@ -14,7 +14,10 @@ public void TypesInInternalsNamespace_AreNotPublic()
 
         Assert.All(types, type =>
         {
-            Assert.False(type.IsPublic, $"Expected type {type.Name} to be internal.");
+            if (type.Name != "HttpTransport")
```

Comment:
@mattjohnsonpint should we move this to `Sentry.Http` namespace (or other) or really open the door to have public types under the `Internal` namespace to 'signify' this won't follow semver. 

ASP.NET Core has public types under `Internal` namespaces, I know as I've relied on them without noticying in the past :D